src/Parameter.cpp

```diff
@@ -129,6 +129,42 @@ bool Parameter::is_buffer() const {
     return contents.ptr->is_buffer;
 }
 
+Expr Parameter::get_scalar_expr() const {
+    check_is_scalar();
+    const Type t = type();
+    if (t.is_bool()) {
+      return Expr(get_scalar<bool>());
+    } else if (t.is_float()) {
+      switch (t.bits) {
+        case 32: return Expr(get_scalar<float>());
+        // Note that we lose precision here, since Expr can only
+        // represent 32-bit ints or floats.
+        case 64: return Expr((float) get_scalar<double>());
+      }
+    } else if (t.is_int()) {
+      switch (t.bits) {
+        case 8: return Expr(get_scalar<int8_t>());
+        case 16: return Expr(get_scalar<int16_t>());
+        case 32: return Expr(get_scalar<int32_t>());
+        // Note that we lose precision here, since Expr can only
+        // represent 32-bit ints or floats.
+        case 64: return Expr((float) get_scalar<int64_t>());
+      }
+    } else if (t.is_uint()) {
+      switch (t.bits) {
+        case 8: return Expr(get_scalar<uint8_t>());
+        case 16: return Expr(get_scalar<uint16_t>());
+        // Fudge this a little bit: Expr handles 32-bit signed ints;
+        // cast uint32 to int32 with the assumption the destination
+        // will do the right thing.
+        case 32: return Expr((int32_t) get_scalar<uint32_t>());
+        // Note that we lose precision here, since Expr can only
+        // represent 32-bit ints or floats.
+        case 64: return Expr((float) get_scalar<uint64_t>());
```

Comment:
Consider checking at runtime if the value is representable by the narrower type, and if not use a more complex piece of IR.